pkg/tcpip/network/fragmentation/fragmentation.go

```diff
@@ -83,7 +88,7 @@ func NewFragmentation(highMemoryLimit, lowMemoryLimit int, reassemblingTimeout t
 
 // Process processes an incoming fragment belonging to an ID
 // and returns a complete packet when all the packets belonging to that ID have been received.
-func (f *Fragmentation) Process(id uint32, first, last uint16, more bool, vv buffer.VectorisedView) (buffer.VectorisedView, bool, error) {
+func (f *Fragmentation) Process(id uint32, first, last uint16, more bool, vv buffer.VectorisedView, headerView buffer.View) (buffer.VectorisedView, bool, error) {
```

Comment:
I believe we should just rewrite this take a PacketBuffer which carries the header rather than add it here as a parameter.

Reply:
Actually PacketBuffer is getting trimmed 20 bytes before calling Process, so we need to store the it in another variable. And that variable we are passing to the Process.